interface/forms/questionnaire_assessments/save.php

```diff
@@ -14,32 +14,71 @@
  * If admin want to register a new form then one is added to forms registry.
  * If already exists then inform user and redirect back to New Questionnaire form.
 */
+require_once(__DIR__ . "/../../../src/Common/Forms/CoreFormToPortalUtility.php");
+
+use OpenEMR\Common\Csrf\CsrfUtils;
+use OpenEMR\Common\Forms\CoreFormToPortalUtility;
+use OpenEMR\Services\QuestionnaireResponseService;
+use OpenEMR\Services\QuestionnaireService;
+
+$isPortal = CoreFormToPortalUtility::isPatientPortalSession($_GET);
+if ($isPortal) {
+    $ignoreAuth_onsite_portal = true;
+}
+$patientPortalOther = CoreFormToPortalUtility::isPatientPortalOther($_GET);
 
 require_once(__DIR__ . "/../../globals.php");
 require_once("$srcdir/api.inc");
 require_once("$srcdir/forms.inc");
 
-use OpenEMR\Common\Csrf\CsrfUtils;
-use OpenEMR\Services\QuestionnaireService;
-
 if (!CsrfUtils::verifyCsrfToken($_POST["csrf_token_form"])) {
     CsrfUtils::csrfNotVerified();
 }
 
```

Comment:
may need to incorporate something to ensure patient doesn't modify another patient form, which can see where done in sdoh here:
https://github.com/openemr/openemr/blob/master/interface/forms/sdoh/save.php#L44-L45